game/src/contrib/systems/HeroUISystem.java

```diff
@@ -0,0 +1,95 @@
+package contrib.systems;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.scenes.scene2d.Group;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.ProgressBar;
+import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable;
+import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
+
+import contrib.components.XPComponent;
+
+import core.Entity;
+import core.Game;
+import core.System;
+import core.components.PlayerComponent;
+import core.components.UIComponent;
+import core.hud.UITools;
+
+import java.util.HashMap;
+
+public final class HeroUISystem extends System {
+    private final HashMap<Entity, HeroUI> map;
+
+    private record HeroUI(ProgressBar pb, Label level, Entity ui) {}
+
+    public HeroUISystem() {
+        super(XPComponent.class, PlayerComponent.class);
+        map = new HashMap<>();
+        onEntityAdd = (x) -> map.put(x, createNewHeroUI(x));
+        onEntityRemove = (x) -> Game.removeEntity(map.remove(x).ui());
+    }
+
+    @Override
+    public void execute() {
+        entityStream().forEach(this::update);
+    }
+
+    private void update(Entity x) {
+        XPComponent xc = x.fetch(XPComponent.class).orElseThrow();
```

Comment:
```suggestion
        XPComponent xc = x.fetch(XPComponent.class).orElseThrow();
```
nutz hier bitte die Factory-Methode in `MissingComponentException`. damit die Fehlermeldung hilfreich ist